### ISSUE TITLE
Identifying coordinate variables

### ISSUE THREAD

**Issue by TomNicholas** [zarr-developers/VirtualiZarr#105]
I'm a little unclear how xarray / netCDF coordinate variables are distinguished from data variables in the Zarr model.

Xarray follows the netCDF convention that 1-dimensional variables with the same name as their only dimension are to be treated as "coordinates", but it is also possible to have additional 1D or multi-dimensional coordinates (which [netCDF calls "auxiliary coordinate variables"](https://www.unidata.ucar.edu/software/netcdf/workshops/2010/datamodels/NcCVars.html)).

I think at the very least we might need an additional line of code inside `open_virtual_dataset` that looks for any metadata attribute named `'coordinates'`.

**Comment by TomNicholas** [comment]
Turns out the way coordinates are specified in the kerchunk references format is as follows:

- If there is one non-dimension coordinate its name is saved as a string in the top-level `.zattrs`, i.e. `'.zattrs': '{"coordinates":"lat"}'`.
- If there is more than one non-dimension coordinate their names are saved not as a list of strings, but as just strings with spaces between them, i.e. `'.zattrs': '{"coordinates":"lat lon"}'`. Note this is a different syntax than how multiple array dimensions are recorded: `'{"_ARRAY_DIMENSIONS":["x","y"]}'`.
- Dimension coordinates are not specified at all, so 1D variables with the same name as their only dimension will be interpreted as coordinates even when they don't have an entry in the `.zattrs`.


(None of this is actually described in the [kerchunk references specification](https://fsspec.github.io/kerchunk/spec.html), I had to work this out by trial and error.)

**Comment by dcherian** [comment]
These are CF conventions and will/should be handled by Xarray

**Comment by TomNicholas** [comment]
> These are CF conventions

Oh okay, I didn't realise that. I'm struggling to find a self-contained description of this in the CF conventions doc but there is at least one example just above [here](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#data-model-cell-measure).

> will/should be handled by Xarray.

Currently because we build the virtual dataset variable-by-variable, and already have to do some sneaky stuff with the coordinates to avoid accidentally creating indexes (inside [`separate_coords`](https://github.com/zarr-developers/VirtualiZarr/blob/ef2429a2001ff3a26a357151913039a5c6d556b3/virtualizarr/xarray.py#L301), I think this has to be done manually within virtualizarr. I've implemented that in #156. It would be nice to delegate this to an xarray function though!

> There's an open issue about implementing `Dataset.encode_cf` that might be useful here

Found it: https://github.com/pydata/xarray/issues/4412

> And for your first question, Zarr doesn't distinguish between coordinate and data variables.

So IIUC in a Zarr store coordinates would be specified to xarray in the same way, through this CF-convention format?
